core/state.go

```diff
@@ -163,17 +163,24 @@ func (s *State) globalTrie(bucket db.Bucket, newTrie trie.NewTrieFunc) (*trie.Tr
 
 	// fetch root key
 	rootKeyDBKey := dbPrefix
-	var rootKey *trie.BitArray // TODO: use value instead of pointer
-	err := s.txn.Get(rootKeyDBKey, func(val []byte) error {
-		rootKey = new(trie.BitArray)
-		return rootKey.UnmarshalBinary(val)
+	var val []byte
+	err := s.txn.Get(rootKeyDBKey, func(data []byte) error {
+		val = data
+		return nil
 	})
```

Comment:
Is any other way of getting the data, or we are always forced to write functions like this, even for the simplest of operations? Out of curiosity

Reply:
I don't think there is other way - https://github.com/NethermindEth/juno/pull/2661#discussion_r2012005225